src/frontend/screens/Track/index.tsx

```diff
@@ -0,0 +1,160 @@
+import {NativeStackNavigationOptions} from '@react-navigation/native-stack';
+
+const m = defineMessages({
+  title: {
+    id: 'screens.Track.title',
+    defaultMessage: 'Track',
+    description:
+      'Title of track screen showing (non-editable) view of observation with map',
+  },
+  deleteTitle: {
+    id: 'screens.Track.deleteTitle',
+    defaultMessage: 'Delete track?',
+    description: 'Title of dialog asking confirmation to delete a track',
+  },
+  tracks: {
+    id: 'screens.Track.tracks',
+    defaultMessage: 'Tracks',
+  },
+});
+
+export const TrackScreen: NativeNavigationComponent<'Track'> = ({
+  route,
+  navigation,
+}) => {
+  const isMine = false;
+  React.useLayoutEffect(() => {
+    navigation.setOptions({
+      headerRight: isMine
+        ? () => (
+            <TouchableOpacity
+              hitSlop={{top: 16, right: 16, bottom: 16, left: 16}}
+              onPress={() => {
+                // TODO: navigate to edit track screen
+              }}>
+              <EditIcon />
+            </TouchableOpacity>
+          )
+        : undefined,
+    });
+  }, [navigation, isMine]);
+
+  const {data: track} = useTrack(route.params.trackId);
+  const {data: observations} = useObservations();
+  const trackObservations = observations.filter(observation =>
+    track.refs.some(ref => ref.id === observation.docId),
+  );
+
+  const deleteTrackMutation = useDeleteTrackMutation();
+
+  function deleteTrack() {
+    deleteTrackMutation.mutate(track.docId, {
+      onSuccess: () => {
+        navigation.pop();
+      },
+    });
+  }
+
+  return (
+    <SafeAreaView style={styles.root}>
+      <ScreenContentWithDock
+        contentContainerStyle={{padding: 0}}
+        dockContainerStyle={{padding: 0}}
+        dockContent={
+          <ActionButtons
+            handleDelete={deleteTrack}
+            isMine={true}
+            deleteMessage={m.deleteTitle}
+          />
+        }>
+        <View>
+          <MapPreview
+            locationHistory={track.locations.map(({timestamp, coords}) => ({
+              latitude: coords.latitude,
+              longitude: coords.longitude,
+              timestamp: parseInt(timestamp, 10),
+            }))}
+            observations={trackObservations}
+          />
+          <View style={styles.trackTitleWrapper}>
+            <TrackIcon />
+            <Text style={styles.trackTitle}>
+              <FormattedMessage {...m.tracks} />
+            </Text>
+          </View>
+          <View style={styles.divider} />
+          <ObservationList observations={trackObservations} />
+          <View style={styles.divider} />
+          <Text style={styles.text}>{track.tags.notes}</Text>
+        </View>
+      </ScreenContentWithDock>
+      <ErrorBottomSheet
+        error={deleteTrackMutation.error}
+        clearError={deleteTrackMutation.reset}
+        tryAgain={deleteTrack}
+      />
+    </SafeAreaView>
+  );
+};
+TrackScreen.navTitle = m.title;
```

Comment:
nit: no longer needed

Reply:
I just went back to standard header. Which means that there is no need to abstract all the navoptions, as the only changing nav option is the title, which is now being handled by the appscreen.tsx